community/index/src/main/java/org/neo4j/index/internal/gbptree/GBPTree.java

```diff
@@ -771,10 +762,10 @@ private void checkpoint( IOLimiter ioLimiter, Header.Writer headerWriter ) throw
 
         // Block writers, or if there's a current writer then wait for it to complete and then block
         // From this point and till the lock is released we know that the tree won't change.
-        long stamp = cleanerLock.writeLock();
-        writerLock.lock();
+        lock.writerAndCleanerLock();
         try
         {
+            assertRecoveryCleanSuccessful();
```

Comment:
on another side, it looks like `cleaning != null` check inside check is redundant